### PR TITLE
feat: add API_KEY list support

### PR DESCRIPTION
**Pull request by neboman11**
Description

This PR fixes https://github.com/go-skynet/LocalAI/issues/729 and https://github.com/go-skynet/LocalAI/issues/480

This PR is a copy of https://github.com/go-skynet/LocalAI/pull/740 but with the API key option changed to a list (as requested in that PR).

This PR adds `api-key` flag. When this flag is set, all API requests needs to provide the same value as Bear token in the request hear. If this flag is not set, no auth will be enabled. This flag is not set by default.

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -130,6 +130,11 @@ func main() {
 				EnvVars: []string{"UPLOAD_LIMIT"},
 				Value:   15,
 			},
+			&cli.StringFlag{
```
[the remark this answers is not in the document]
Comment (neboman11) [reply main.go]: Changed it to use the StringSliceFlag

**Review by mudler — APPROVED** [review]
Looking good here, thanks!

**Comment by luisfarzati** [comment]
@neboman11 you rock ❤️ thank you!

**Comment by alexander-potemkin** [comment]
I'm sorry, and how does it work with `models/available` endpoint?

I'm having problems checking for the models available:

```bash
# curl http://localhost:8080/models/available -H "Authorization: Bearer $myApiKey"
{"message":"Invalid API key"}
```

**Comment by alexander-potemkin** [comment]
btw, same when I try to query the model, by using default's OpenAI's example:

```bash
curl http://localhost:8080/v1/chat/completions   -H "Content-Type: application/json"   -H "Authorization: Bearer $myApiKey"   -d '{
    "model": "gpt-3.5-turbo",
    "messages": [
      {
        "role": "system",
        "content": "You are a poetic assistant, skilled in explaining complex programming concepts with creative flair."
      },
      {
        "role": "user",
        "content": "Compose a poem that explains the concept of recursion in programming."
      }
    ]
  }'
{"message":"Invalid API key"}
```

**Comment by alexander-potemkin** [comment]
Please, forgive me - I shouldn't been adding quotes for the key in my `docker-compose.yml`.

**Pull request merged**